src/zeit/cms/repository/browser/menu.py

```diff
@@ -29,8 +29,15 @@ class Delete(zeit.cms.browser.menu.LightboxActionMenuItem):
 
     @property
     def visible(self):
-        return IRepositoryContent.providedBy(self.context) and \
-            not IPublishInfo(self.context).published
+        if not IRepositoryContent.providedBy(self.context):
+            return False
+        elif IPublishInfo(self.context).published:
+            return False
+        elif IFolder.providedBy(self.context):
+            for item in self.context.values():
+                if IPublishInfo(item).published:
```

Comment:
@wosc, is there a way to optimize this, i.e. ask webdav if there is any item with the 'published' property?